src/test/java/edu/cornell/kfs/gl/service/impl/ScheduledAccountingLineServiceImplTest.java

```diff
@@ -1,113 +1,119 @@
-		assertEquals("Date is not what we expected.", expectedDate.getDate(), results.getDate());
-		assertEquals("Month is not what we expected.", expectedDate.getMonth(), results.getMonth());
-		assertEquals("Year is not what we expected.", expectedDate.getYear(), results.getYear());
-	}
-	
-	private Date getExpectedDate(int dateAddElement, int addAmount) {
-		Calendar calcCal = startingPointCal.getInstance();
-		calcCal.add(dateAddElement, addAmount);
-		Date expectedDate = new Date(calcCal.getTimeInMillis());
-		return expectedDate;
-	}
-	
-	private Date getDateResult(ScheduleTypes scheduleType, String count) {
-		accountingLine.setScheduleType(scheduleType.name);
-		accountingLine.setPartialTransactionCount(count);;
-		Date results = scheduledAccountingLineService.generateEndDate(accountingLine);
-		return results;
-	}
+
+    ScheduledAccountingLineServiceImpl scheduledAccountingLineService;
+    Date startingPointDate;
+    Calendar startingPointCal;
+    ScheduledSourceAccountingLine accountingLine;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        Configurator.setLevel(ScheduledAccountingLineServiceImpl.class.getName(), Level.DEBUG);
+        scheduledAccountingLineService = new ScheduledAccountingLineServiceImpl();
+
+        startingPointDate = new Date(Calendar.getInstance().getTimeInMillis());
+        startingPointCal = Calendar.getInstance();
+        startingPointCal.setTimeInMillis(startingPointDate.getTime());
+
+        accountingLine = new TestableScheduledSourceAccountingLine();
+        accountingLine.setStartDate(startingPointDate);
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        scheduledAccountingLineService = null;
+        startingPointDate = null;
+        startingPointCal = null;
+        accountingLine = null;
+        GlobalVariables.getMessageMap().clearErrorMessages();
+    }
+
+    @ParameterizedTest
+    @EnumSource(ScheduledAccountingLineServiceImplEndDateFixture.class)
+    void testGenerateEndDates(ScheduledAccountingLineServiceImplEndDateFixture fixture) {
+        doTest(fixture.expectedCalendarType, fixture.expectedAmount, fixture.resultType, fixture.resultAmount);
+    }
+
+    private void doTest(int expectedCalendarType, int expectedAmount,
+            CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes resultType, String resultAmount) {
+        Date expectedDate = getExpectedDate(expectedCalendarType, expectedAmount);
+        Date results = getDateResult(resultType, resultAmount);
+
+        assertEquals("Day is not what we expected.", expectedDate.getDay(), results.getDay());
+        assertEquals("Date is not what we expected.", expectedDate.getDate(), results.getDate());
+        assertEquals("Month is not what we expected.", expectedDate.getMonth(), results.getMonth());
+        assertEquals("Year is not what we expected.", expectedDate.getYear(), results.getYear());
+    }
```

Comment:
All of the "get" methods on lines 65 to 68 for these comparisons are showing warnings indicating they are deprecated. Should we be modifying existing unit tests and adding deprecated methods?